lib/krane/kubernetes_resource.rb

```diff
@@ -134,22 +131,12 @@ def to_kubeclient_resource
 
     def validate_definition(kubectl, selector: nil)
       @validation_errors = []
-      @validation_warnings = []
       validate_selector(selector) if selector
       validate_timeout_annotation
-      validate_annotation_version
       validate_spec_with_kubectl(kubectl)
       @validation_errors.present?
     end
 
-    def validation_warning_msg
-      @validation_warnings.join("\n")
-    end
-
-    def has_warnings?
-      @validation_warnings.present?
```

Comment:
So the only validation warnings we ever generated are from deprecated kubernetes.shopify.io annotations? Are we missing anything else or is that really it?

Reply:
Yeah that's it. I went based on this commit: https://github.com/Shopify/krane/commit/d237e0c80ab36db6d77577858a9de76b6f781b40 where the annotations were first deprecated. The warnings were added to support that.